packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -4671,6 +4672,94 @@ describe('SnapController', () => {
     });
   });
 
+  describe('onClientRequest', () => {
+    it('returns the value when `onClientRequest` returns a valid response', async () => {
+      const rootMessenger = getControllerMessenger();
+      const messenger = getSnapControllerMessenger(rootMessenger);
+      const snapController = getSnapController(
+        getSnapControllerOptions({
+          messenger,
+          state: {
+            snaps: getPersistedSnapsState(),
+          },
+        }),
+      );
+
+      rootMessenger.registerActionHandler(
+        'SubjectMetadataController:getSubjectMetadata',
+        () => MOCK_SNAP_SUBJECT_METADATA,
+      );
+
+      rootMessenger.registerActionHandler(
+        'ExecutionService:handleRpcRequest',
+        async () =>
+          Promise.resolve({
+            foo: 'bar',
+          }),
```

Comment:
Nit: `Promise.resolve` is redundant here.
```suggestion
        async () => ({
            foo: 'bar',
        }),
```